jps-plugin/src/org/intellij/erlang/jps/builder/ErlangModuleBuildOrderBuilder.java

```diff
@@ -0,0 +1,194 @@
+      @Nullable
+      @Override
+      public ErlangTarget fun(BuildTarget<?> buildTarget) {
+        return buildTarget instanceof ErlangTarget ? (ErlangTarget) buildTarget : null;
+      }
+    });
+    for (ErlangTarget erlangTarget : erlangTargets) {
+      erlangTarget.setBuildOrder(new ErlangModuleBuildOrder());
+    }
+
+    LOG.debug("Collect dirty files.");
+    List<String> dirtyErlangFilePaths = new DirtyFileProcessor<String, ErlangModuleBuildOrderTarget>() {
+      @Nullable
+      @Override
+      protected String getDirtyElement(@NotNull File file) throws IOException {
+        String fileName = file.getName();
+        return isSource(fileName) || isHeader(fileName) ? file.getAbsolutePath() : null;
+      }
+    }.collectDirtyElements(holder);
+
+    if (dirtyErlangFilePaths.isEmpty()) {
+      LOG.debug("There aren't dirty .erl or .hrl files.");
+    }
+    else {
+      LOG.debug("Search dirty modules.");
+      List<String> sortedDirtyModules = getSortedDirtyModules(projectBuildOrder, dirtyErlangFilePaths);
+      addFilesToBuiltTarget(context, sortedDirtyModules);
+    }
+
+  }
+
+  @NotNull
+  @Override
+  public String getPresentableName() {
+    return NAME;
+  }
+
+  @Nullable
+  private static ErlangProjectBuildOrder loadProjectBuildOrder(@NotNull CompileContext context) {
+    return readFromXML(context, BUILD_ORDER_FILE_NAME, ErlangProjectBuildOrder.class);
+  }
+
+  @NotNull
+  private static List<String> getSortedDirtyModules(@NotNull ErlangProjectBuildOrder projectBuildOrder,
+                                                    @NotNull List<String> dirtyErlangFilePaths) {
+    Set<String> allDirtyFiles = getAllDirtyFiles(projectBuildOrder, dirtyErlangFilePaths);
+    return getSortedDirtyModules(projectBuildOrder.myErlangFiles, allDirtyFiles);
+  }
+
+  private static void addPrepareDependenciesFailedMessage(@NotNull CompileContext context) {
+    context.processMessage(new CompilerMessage(NAME, BuildMessage.Kind.WARNING, "The project will be fully rebuilt due to errors."));
+  }
+
+  @NotNull
+  private static Set<String> getAllDirtyFiles(@NotNull ErlangProjectBuildOrder projectBuildOrder,
+                                              @NotNull List<String> dirtyFiles) {
+    SortedModuleDependencyGraph graph = new SortedModuleDependencyGraph(projectBuildOrder);
+    Set<String> allDirtyFiles = ContainerUtil.newHashSet();
+    for (String node : dirtyFiles) {
+      findDirtyFiles(node, GraphGenerator.create(graph), allDirtyFiles);
+    }
+    return allDirtyFiles;
+  }
+
+  private static void findDirtyFiles(@NotNull String filePath,
+                                     @NotNull GraphGenerator<String> dependenciesGraph,
+                                     @NotNull Set<String> dirtyFiles) {
+    if (dirtyFiles.contains(filePath)) return;
+    dirtyFiles.add(filePath);
+    Iterator<String> childIterator = dependenciesGraph.getOut(filePath);
```

Comment:
I don't like being THAT pedantic, but I hoped you'd figure it out yourself. You agreed that you can't understand what this method does without it's context. Can you understand what it does now without this variable named properly? How would one know how the dependency graph is built and what does in and out mean?

Just naming it 'dependentFiles' would make the picture complete - we'll be able to read this method and understand that it marks all files dependent on dirty files as dirty. Let me know if I'm wrong.